### PR TITLE
Expand but hide "1 similar comment" in `hide-useless-comments` #3830

### LINKED ISSUE #3830: Expand but hide "1 similar comment" in `hide-useless-comments`
"Similar" comments are useless comments, so they should be hidden by `hide-useless-comments`.

The feature should run on these comments and, if they are indeed useless, the should be expanded ¹ and hidden.

---

¹ the user shouldn't have to click that comment to expand it again if they already clicked "show useless comments"

Test URL: https://github.com/justinfrankel/licecap/issues/97#issuecomment-737063337

### PR DESCRIPTION
**Pull request by PauloCarneiro99**
1. LINKED ISSUES:
   Closes #3830 

2. TEST URLS:
   https://github.com/justinfrankel/licecap/issues/97#issuecomment-737063337

### CODE REVIEW

**Comment by yakov116** [comment]
Thanks for the PR. https://github.com/sindresorhus/refined-github/pull/3827 is blocking and we will need that to be merged first

File: source/features/hide-useless-comments.tsx [hunk 1]
```diff
@@ -15,8 +15,28 @@ function unhide(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
...
+	// Get all "similar comment" boxes
+	const similarCommentBoxes = select.all('.Details-element');
+
+	for (const similarCommentsBox of similarCommentBoxes) {
+		const similarCommentBtn: HTMLElement = similarCommentsBox.querySelectorAll('.Details-content--closed')[0] as HTMLElement;
+		for (const comment of similarCommentsBox.querySelectorAll('.js-timeline-item')) {
+			commentsToHide.push(comment); // Add all similar comments to be hidden
+		}
+
+		similarCommentBtn.click(); // Expand all the similar comments on this box
```
Comment (fregante) [inline source/features/hide-useless-comments.tsx:37]: This should be done only when the user expands the comments

File: source/features/hide-useless-comments.tsx [hunk 2]
```diff
@@ -15,8 +15,28 @@ function unhide(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
...
+
 function init(): void {
-	let uselessCount = 0;
+	const commentsToHide: any[] = [];
+
+	// Get all "similar comment" boxes
+	const similarCommentBoxes = select.all('.Details-element');
+
+	for (const similarCommentsBox of similarCommentBoxes) {
+		const similarCommentBtn: HTMLElement = similarCommentsBox.querySelectorAll('.Details-content--closed')[0] as HTMLElement;
```
Comment (cheap-glitch) [inline source/features/hide-useless-comments.tsx:32]: `select()` can take a root element for the query as the second argument:
```suggestion
		const similarCommentBtn = select('.Details-content--closed', similarCommentsBox);
```
Same for the next line.

Comment (fregante) [reply source/features/hide-useless-comments.tsx:32]: There are several querySelectors while there should only be one at the top (and only one loop). Once my other comment is applied, this should become clearer

**Comment by fregante** [comment]
https://github.com/sindresorhus/refined-github/pull/3827 was merged, this can now continue

**Comment by fregante** [comment]
Closing this for now, but please continue if you have time.

**Pull request closed without merging**